src/interface/blas2_interface.hpp

```diff
@@ -605,6 +605,58 @@ typename sb_handle_t::event_t _syr_impl(
   }
 }
 
+/*! _SPR.
+ * @brief Implementation of the rank 1 operation
+ */
+/*
+sspr 	( 	character  	UPLO,
+   integer  	N,
+   real  	ALPHA,
+   real, dimension(N)  	X,
+   integer  	INCX,
+   real, dimension(N, N + 1 / 2)  	AP
+ )
+*/
+template <typename sb_handle_t, typename index_t, typename element_t,
+          typename container_t0, typename increment_t, typename container_t1>
+typename sb_handle_t::event_t _spr_impl(sb_handle_t& sb_handle, char _Uplo,
+                                        index_t _N, element_t _alpha,
+                                        container_t0 _vx, increment_t _incx,
+                                        container_t1 _mPA) {
+  // throw exception if invalid arguments
+  if (_N <= 0) {
+    throw std::invalid_argument("Invalid vector size");
+  }
+
+  // bail out early if alpha == 0
+  if (_alpha == (element_t)0) {
+    std::vector<cl::sycl::event> event;
+    return event;
+  }
+
+  typename sb_handle_t::event_t ret;
+  _Uplo = tolower(_Uplo);
+  const int Upper = _Uplo == 'u';
+  auto mA = make_matrix_view<col_major>(_mPA, _N, (_N + 1) / 2, _N);
+  auto vx = make_vector_view(_vx, std::abs(_incx), _N);
```

Comment:
Do we really need the `abs` value of `_incx`?

Reply:
This `std::abs` was required because I was calculating the final indices myself inside the kernel rather than using the vector view. I have updated the kernel to use the vector and matrix view `eval` functions instead and removed this `std::abs`.